js/data.js

```diff
@@ -6,3 +6,33 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+var $img = document.getElementsByTagName('img')[0];
+var placeholder = $img.getAttribute('src');
+var $form = document.querySelector('form');
+var $elements = $form.elements;
+var $entryTitle = $elements.title;
+var $photoUrl = $elements.photoUrl;
+var $entryNotes = $elements.notes;
+
+$form.addEventListener('submit', handleSubmit);
+function handleSubmit(event) {
+  var $newEntry = {};
+  $newEntry[$entryTitle.name] = $entryTitle.value;
+  $newEntry[$photoUrl.name] = $photoUrl.value;
+  $newEntry[$entryNotes.name] = $entryNotes.value;
+  $newEntry.nextEntryId = data.nextEntryId;
+  data.nextEntryId += 1;
+  data.entries.unshift($newEntry);
+  $form.reset();
+  $img.setAttribute('src', placeholder);
+  event.preventDefault();
```

Comment:
I usually prefer that `event.preventDefault()` is the first line of any function it needs to be a part of, to help with readability